sdk/lib/drivers/sound/mmixer/controls.c

```diff
@@ -181,53 +182,44 @@ MMixerAddMixerControl(
 
             DPRINT("NodeIndex %u Range Min %d Max %d Steps %x UMin %x UMax %x\n", NodeIndex, Range->Bounds.SignedMinimum, Range->Bounds.SignedMaximum, Range->SteppingDelta, Range->Bounds.UnsignedMinimum, Range->Bounds.UnsignedMaximum);
 
-            MaxRange = Range->Bounds.UnsignedMaximum - Range->Bounds.UnsignedMinimum;
+            VolumeData = (LPMIXERVOLUME_DATA)MixerContext->Alloc(sizeof(*VolumeData));
+            if (!VolumeData)
+                return MM_STATUS_NO_MEMORY;
+
+            /* store mixer control info there */
+            VolumeData->Header.dwControlID = MixerControl->Control.dwControlID;
+            VolumeData->SignedMinimum = Range->Bounds.SignedMinimum;
+            VolumeData->SignedMaximum = Range->Bounds.SignedMaximum;
+            VolumeData->SteppingDelta = Range->SteppingDelta;
 
-            if (MaxRange)
+            /* Fallback to defaults if: 1) the range is not defined (typically is 0) */
+            if (VolumeData->SignedMinimum == VolumeData->SignedMaximum)
             {
-                ASSERT(MaxRange);
-                VolumeData = (LPMIXERVOLUME_DATA)MixerContext->Alloc(sizeof(MIXERVOLUME_DATA));
-                if (!VolumeData)
-                    return MM_STATUS_NO_MEMORY;
-
-                Steps = MaxRange / Range->SteppingDelta + 1;
-
-                /* store mixer control info there */
-                VolumeData->Header.dwControlID = MixerControl->Control.dwControlID;
-                VolumeData->SignedMaximum = Range->Bounds.SignedMaximum;
-                VolumeData->SignedMinimum = Range->Bounds.SignedMinimum;
-                VolumeData->SteppingDelta = Range->SteppingDelta;
-                VolumeData->ValuesCount = Steps;
-                VolumeData->InputSteppingDelta = 0x10000 / Steps;
-
-                VolumeData->Values = (PLONG)MixerContext->Alloc(sizeof(LONG) * Steps);
-                if (!VolumeData->Values)
-                {
-                    MixerContext->Free(Desc);
-                    MixerContext->Free(VolumeData);
-                    return MM_STATUS_NO_MEMORY;
-                }
+                VolumeData->SignedMinimum = -96 * 0x10000; // -96 DB
+                VolumeData->SignedMaximum = 0; // 0 DB
+            }
 
-                Value = Range->Bounds.SignedMinimum;
-                for (Index = 0; Index < Steps; Index++)
-                {
-                    VolumeData->Values[Index] = Value;
-                    // HACK: use '- 1' to make the left and right volume controls behave independently.
-                    Value += Range->SteppingDelta - 1;
-                }
-                MixerControl->ExtraData = VolumeData;
-           }
-       }
-       MixerContext->Free(Desc);
+            MixerControl->ExtraData = VolumeData;
+        }
+        else
+        {
+            VolumeData = (LPMIXERVOLUME_DATA)MixerContext->Alloc(sizeof(*VolumeData));
+            if (!VolumeData)
+                return MM_STATUS_NO_MEMORY;
```

Comment:
So you can remove this one (it can be factored outside of the if, see comment above).

Reply:
Committed. See my new comment in the first place.